homeassistant/components/device_tracker/tesla.py

```diff
@@ -0,0 +1,60 @@
+"""
+Support for the Tesla platform.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.tesla/
+"""
+import logging
+
+from homeassistant.components.tesla import DOMAIN as TESLA_DOMAIN
+from homeassistant.helpers.event import track_utc_time_change
+from homeassistant.util import slugify
+
+DEPENDENCIES = ['tesla']
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_scanner(hass, config, see, discovery_info=None):
+    """Set up the Tesla tracker."""
+    TeslaDeviceTracker(
+        hass, config, see,
+        hass.data[TESLA_DOMAIN]['devices']['devices_tracker'])
+    return True
+
+
+class TeslaDeviceTracker(object):
+    """A class representing a Tesla device."""
+
+    def __init__(self, hass, config, see, tesla_devices, ):
+        """Initialize the Tesla device scanner."""
+        self.hass = hass
+        self.see = see
+        self.devices = tesla_devices
+        self._update_info()
+
+        track_utc_time_change(
+            self.hass, self._update_info, second=range(0, 60, 30))
+
+    def _update_info(self, now=None):
+        """Update the device info."""
+        for device in self.devices:
+            device.update()
+            name = device.name
+            _LOGGER.debug('Updating device position: %s', name)
+            tesla_id = slugify(device.uniq_name)
+            dev_id = slugify(name)
+
+            if dev_id is None:
```

Comment:
`dev_id` can't be `None` after return from `slugify`. It will be a string. Otherwise it will be an error. If `name` can be `None` you need to check for that before passing it to `slugify`.

Reply:
It can't be None.
I've removed this part because dev_id is not used anywhere else. I didn't remember why I've added it...